/gsd auto stuck at attempting to recreate a missing summary

### GSD version

2.22.0

### Affected area

Auto-mode / dispatch loop

### What happened?

/gsd auto Slice 4 of Milestone 1 completed and was unable to continue

Tried /gsd doctor audit and heal

It attempted a fix. Resulting in a loop.



### Expected behavior

Issue seems to be stated clearly but there is a looping state while attempting to resolve.

### Steps to reproduce

[pacman.zip](https://github.com/user-attachments/files/26040606/pacman.zip)

run with gsd.
/gsd auto

It will demonstrate the oscillating messages.

### Error output / logs

```shell
Warning: GSD doctor audit.
 Scope: all milestones
 Issues: 2 total · 1 error(s) · 1 warning(s) · 1 fixable
 Top issue types:
 - state_file_stale: 1
 - task_done_missing_summary: 1
 Priority issues:
 - [WARN] project: STATE.md is stale — shows "summarizing" but derived state is "planning" (.gsd/STATE.md)
 - [ERROR] M001/S01/T01: Task T01 is marked done but summary is missing
 (.gsd/milestones/M001/slices/S01/tasks/T01-SUMMARY.md)

 Doctor heal found nothing actionable to hand off to the LLM.

 Warning: GSD doctor audit.
 Scope: all milestones
 Issues: 1 total · 1 error(s) · 0 warning(s) · 0 fixable
 Top issue types:
 - task_done_missing_summary: 1
 Priority issues:
 - [ERROR] M001/S01/T01: Task T01 is marked done but summary is missing
 (.gsd/milestones/M001/slices/S01/tasks/T01-SUMMARY.md)

 Skipped 20+ completed units. Yielding to UI before continuing.


The last line alternates between the "Skipped... " above and:
Skipping complete-slice M001/S04 — already completed in a prior session. Advancing.
```

### Operating system

macOS

### Node.js version

_No response_

### AI provider (if relevant)

OpenAI-compatible

### Additional context

_No response_

--- a/src/resources/extensions/gsd/auto-worktree.ts
+++ b/src/resources/extensions/gsd/auto-worktree.ts
@@ -158,7 +158,15 @@ export function createAutoWorktree(basePath: string, milestoneId: string): strin
   // Planning artifacts may be untracked if the project's .gitignore had a
   // blanket .gsd/ rule (pre-v2.14.0). Without this copy, auto-mode loops
   // on plan-slice because the plan file doesn't exist in the worktree.
-  copyPlanningArtifacts(basePath, info.path);
+  //
+  // IMPORTANT: Skip when re-attaching to an existing branch (#759).
+  // The branch checkout already has committed artifacts with correct state
+  // (e.g. [x] for completed slices). Copying from the project root would
+  // overwrite them with stale data ([ ] checkboxes) because the root is
+  // not always fully synced.
+  if (!branchExists) {
+    copyPlanningArtifacts(basePath, info.path);
+  }
 
   // Run user-configured post-create hook (#597) — e.g. copy .env, symlink assets
   const hookError = runWorktreePostCreateHook(basePath, info.path);

--- a/src/resources/extensions/gsd/state.ts
+++ b/src/resources/extensions/gsd/state.ts
@@ -32,7 +32,6 @@ import {
 
 import { milestoneIdSort, findMilestoneIds } from './guided-flow.js';
 import { nativeBatchParseGsdFiles, type BatchParsedFile } from './native-parser-bridge.js';
-import { isDbAvailable, _getAdapter } from './gsd-db.js';
 
 import { join, resolve } from 'path';
 import { debugCount, debugTime } from './debug-logger.js';
@@ -149,37 +148,18 @@ async function _deriveStateImpl(basePath: string): Promise<GSDState> {
   const fileContentCache = new Map<string, string>();
   const gsdDir = gsdRoot(basePath);
 
-  // ── DB-first content loading ──
-  // When the DB is available, load artifact content from the artifacts table
-  // (indexed SELECT instead of O(N) file I/O). Falls back to native Rust batch
-  // parser, which in turn falls back to sequential JS reads via cachedLoadFile.
-  let dbContentLoaded = false;
-  if (isDbAvailable()) {
-    const adapter = _getAdapter();
-    if (adapter) {
-      try {
-        const rows = adapter.prepare('SELECT path, full_content FROM artifacts').all();
-        for (const row of rows) {
-          const relPath = (row as Record<string, unknown>)['path'] as string;
-          const content = (row as Record<string, unknown>)['full_content'] as string;
-          const absPath = resolve(gsdDir, relPath);
-          fileContentCache.set(absPath, content);
-        }
-        dbContentLoaded = rows.length > 0;
-      } catch {
-        // DB query failed — fall through to native batch parse
-      }
-    }
-  }
-
-  if (!dbContentLoaded) {
+  // NOTE: We intentionally do NOT load from the SQLite DB here (#759).
+  // The DB's artifacts table is populated once during migrateFromMarkdown
+  // and is never updated when files change on disk (e.g. roadmap [x] updates,
+  // plan checkbox changes). Using stale DB content causes deriveState to
+  // return incorrect phase/slice state, leading to infinite skip loops.
+  // The native Rust batch parser is fast enough for state derivation.
   const batchFiles = nativeBatchParseGsdFiles(gsdDir);
   if (batchFiles) {
     for (const f of batchFiles) {
       const absPath = resolve(gsdDir, f.path);
       fileContentCache.set(absPath, f.rawContent);
     }
-  }
   }
 
   /**

--- a/src/resources/extensions/gsd/tests/derive-state-db.test.ts
+++ b/src/resources/extensions/gsd/tests/derive-state-db.test.ts
@@ -248,31 +248,24 @@ async function main(): Promise<void> {
     }
   }
 
-  // ─── Test 5: Requirements counting from DB content ────────────────────
-  console.log('\n=== derive-state-db: requirements from DB content ===');
+  // ─── Test 5: Requirements counting from disk (DB no longer used for content) ─
+  console.log('\n=== derive-state-db: requirements from disk content ===');
   {
     const base = createFixtureBase();
     try {
       // Write minimal milestone dir (needed for milestone discovery)
       mkdirSync(join(base, '.gsd', 'milestones', 'M001'), { recursive: true });
-      // Do NOT write REQUIREMENTS.md to disk — only in DB
-
-      openDatabase(':memory:');
-      insertArtifactRow('REQUIREMENTS.md', REQUIREMENTS_CONTENT, {
-        artifact_type: 'requirements',
-      });
+      // Write REQUIREMENTS.md to disk (DB content is no longer used by deriveState)
+      writeFile(base, 'REQUIREMENTS.md', REQUIREMENTS_CONTENT);
 
       invalidateStateCache();
       const state = await deriveState(base);
 
-      // Requirements should come from DB
-      assertEq(state.requirements?.active, 2, 'req-from-db: requirements.active = 2');
-      assertEq(state.requirements?.validated, 1, 'req-from-db: requirements.validated = 1');
-      assertEq(state.requirements?.total, 3, 'req-from-db: requirements.total = 3');
-
-      closeDatabase();
+      // Requirements should come from disk
+      assertEq(state.requirements?.active, 2, 'req-from-disk: requirements.active = 2');
+      assertEq(state.requirements?.validated, 1, 'req-from-disk: requirements.validated = 1');
+      assertEq(state.requirements?.total, 3, 'req-from-disk: requirements.total = 3');
     } finally {
-      closeDatabase();
       cleanup(base);
     }
   }

